markdown/3.x/en/ff-filter-cloud.md

```diff
@@ -0,0 +1,99 @@
+
+
+### Rendered HTML
+When setup like in the example above, the rendered HTML could look like this:
+```html
+<ff-filter-cloud>
+    <span data-template="filter">Element 1</span>
+    <span data-template="filter">Element 2</span>
+    <span data-template="filter">Element 3</span>
+    <span data-template="filter">Element 4</span>
+</ff-filter-cloud>
+```
+
+## Adding a filter template
+To customize filter element appearance, you can add an HTML element annotated with `[data-template-filter]`. You can access the filter group's name through `{{group.name}}`:
+```html
+<ff-filter-cloud>
+    <span data-template="filter">{{group.name}}: {{element.name}}</span>
+</ff-filter-cloud>
+```
+
+### Rendered HTML
+When setup like in the example above, the rendered HTML could look like this:
+```html
+<ff-filter-cloud>
+    <span data-template="filter">Group 1: Element 1</span>
+    <span data-template="filter">Group 2: Element 2</span>
+    <span data-template="filter">Group 3: Element 3</span>
+    <span data-template="filter">Group 4: Element 4</span>
+</ff-filter-cloud>
+```
+
+Custom filter template can be used with some additional custom HTML:
+```html
+<ff-filter-cloud>
+    <h4>The Filter Cloud</h4>
+
+    <div>
+        <div>Custom above</div>
+        
+        <div data-template="filter" class="filter-element">
+            <span>{{group.name}}: {{element.name}}</span><span> ×</span>
+        </div>
+        
+        <div>Custom below</div>
+  </div>
+</ff-filter-cloud>
+```
+
+### Rendered HTML
+When setup like in the example above, the rendered HTML could look like this:
+```html
+<ff-filter-cloud>
+    <h4>The Filter Cloud</h4>
+
+    <div>
+        <div>Custom above</div>
+        
+        <div data-template="filter" class="filter-element">
+            <span>Group 1: Element 1</span><span> ×</span>
+        </div>
+        <div data-template="filter" class="filter-element">
+            <span>Group 2: Element</span><span> ×</span>
+        </div>
+        <div data-template="filter" class="filter-element">
+            <span>Group 3: Element 3</span><span> ×</span>
+        </div>
+
+        <div>Custom below</div>
+    </div>
+</ff-filter-cloud>
+```
+
+**NOTE** If no element annotated with `[data-template-filter]` is specified, a console warning will be displayed and any existing HTML will be replaced with default template. This also applies to default `ff-filter-could` setup.
+
+## Filter click
+There is only one delegated `click` listener for the filter elements and if is attached to the `ff-filter-cloud` itself. Click events on filter items bubble up to `ff-filter-cloud` 
+and are processed there. By default, clicking on the filter removes it. Users can attach their own `click` and set `e.stopPropagation()` to prevent default behavior. 
+See the example below that shows how to create a filter element with clickable 'x' button:
+```html
```

Comment:
Simplifying this example would be beneficial to users I think. For example:
```html
<script>
    function userOnClick(event) {
        if (!event.target.classList.contains("btn-deselect")) {
            event.stopPropagation();
        }
    }
</script>
<ff-filter-cloud unresolved>
    <div data-template="filter" style="cursor: default" onclick="userOnClick(event)">
        <span style="cursor: default">{{element.name}}</span> <span class="btn-deselect" style="cursor: pointer">×</span>
    </div>
</ff-filter-cloud>
```

Reply:
OK, I'll use it.